From dc164d1e6c76e52d091d20b147fcda50acc5b0c6 Mon Sep 17 00:00:00 2001
From: xx <xx@xx.com>
Date: Wed, 10 Jul 2019 06:46:00 +0000
Subject: [PATCH] disable memory management

---
 src/rtpconfig.h.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rtpconfig.h.in b/src/rtpconfig.h.in
index ea95ecf..cbf23a8 100644
--- a/src/rtpconfig.h.in
+++ b/src/rtpconfig.h.in
@@ -7,7 +7,7 @@
 
   This library was developed at the Expertise Centre for Digital Media
   (http://www.edm.uhasselt.be), a research center of the Hasselt University
-  (http://www.uhasselt.be). The library is based upon work done for 
+  (http://www.uhasselt.be). The library is based upon work done for
   my thesis at the School for Knowledge Technology (Belgium/The Netherlands).
 
   Permission is hereby granted, free of charge, to any person obtaining a
@@ -77,7 +77,7 @@ ${RTP_SUPPORT_IFADDRS}
 
 ${RTP_SUPPORT_SENDAPP}
 
-${RTP_SUPPORT_MEMORYMANAGEMENT}
+// ${RTP_SUPPORT_MEMORYMANAGEMENT}
 
 ${RTP_SUPPORT_RTCPUNKNOWN}
 
-- 
2.7.4

