From b85a29e23aa22f268b5d38c8683cc2cdc0dc79c7 Mon Sep 17 00:00:00 2001
From: zhoubo <syuuha@qq.com>
Date: Thu, 17 Feb 2022 11:10:46 +0800
Subject: [PATCH] Fix-to-use-the-stdc++-instead-of-pthread-rt in android

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f05adb3..beb409b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -667,7 +667,8 @@ if(UNIX)
     elseif(QNXNTO)
       set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} m)
     else()
-      set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} dl m pthread rt)
+      set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} dl m log)
+      # set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} dl m pthread rt)
     endif()
   else()
     set(HAVE_PTHREAD 1)
-- 
2.27.0.221.ga08a83db2b

